### PR TITLE
Internal channels for scheduler

### LINKED ISSUE #84338: Introduce Internal Channels for scheduler
**What would you like to be added**:
The `scheduler` currently has a generic `util` package that stores a lot of common functionalities. Currently the `error_channels` for any errors that may arise in the scheduling framework are present here. W.r.t #71863 which serves as umbrella issue for scheduler cleanup, channels should become an internal package for scheduler.

**Why is this needed**:
The `error_channel` should be a scheduler exclusive package and must not be accessible to other components.

### PR DESCRIPTION
**Pull request by hprateek43**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #84338 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go [hunk 1]
```diff
@@ -150,7 +149,8 @@ func autoConvert_v1alpha1_KubeSchedulerConfiguration_To_config_KubeSchedulerConf
 	if err := Convert_v1alpha1_KubeSchedulerLeaderElectionConfiguration_To_config_KubeSchedulerLeaderElectionConfiguration(&in.LeaderElection, &out.LeaderElection, s); err != nil {
 		return err
 	}
-	if err := componentbaseconfigv1alpha1.Convert_v1alpha1_ClientConnectionConfiguration_To_config_ClientConnectionConfiguration(&in.ClientConnection, &out.ClientConnection, s); err != nil {
```
Comment (draveness) [inline pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go:153]: Is this auto generated by the conversion-go, it does not seem relevant to this

**Comment by hprateek43** [comment]
@ahg-g @alculquicondor Can you review this PR?

File: pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go (hunk 1 above)
Comment (hprateek43) [reply pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go]: Yes, this is auto generated

**Comment by hprateek43** [comment]
@alculquicondor @draveness Can you give a `lgtm` ?

**Comment by draveness** [comment]
> @alculquicondor @draveness Can you give a `lgtm` ?

The test is failing, could you please take another look?

**Comment by hprateek43** [comment]
@draveness The tests are failing because of timeouts. I don't think it should stop a `lgtm`

**Comment by alculquicondor** [comment]
Timeout #89847

/lgtm
